src/main/java/org/opensearch/flowframework/rest/RestGetWorkflowAction.java

```diff
@@ -0,0 +1,108 @@
+import static org.opensearch.flowframework.common.CommonValue.WORKFLOW_URI;
+import static org.opensearch.flowframework.common.FlowFrameworkSettings.FLOW_FRAMEWORK_ENABLED;
+
+/**
+ * Rest Action to facilitate requests to get a workflow status
+ */
+public class RestGetWorkflowAction extends BaseRestHandler {
+
+    private static final String GET_WORKFLOW_ACTION = "get_workflow";
+    private static final Logger logger = LogManager.getLogger(RestGetWorkflowAction.class);
+    private FlowFrameworkFeatureEnabledSetting flowFrameworkFeatureEnabledSetting;
+
+    /**
+     * Instantiates a new RestGetWorkflowAction
+     * @param flowFrameworkFeatureEnabledSetting Whether this API is enabled
+     */
+    public RestGetWorkflowAction(FlowFrameworkFeatureEnabledSetting flowFrameworkFeatureEnabledSetting) {
+        this.flowFrameworkFeatureEnabledSetting = flowFrameworkFeatureEnabledSetting;
+    }
+
+    @Override
+    public String getName() {
+        return GET_WORKFLOW_ACTION;
+    }
+
+    @Override
+    protected BaseRestHandler.RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
+
+        try {
+            if (!flowFrameworkFeatureEnabledSetting.isFlowFrameworkEnabled()) {
+                throw new FlowFrameworkException(
+                    "This API is disabled. To enable it, update the setting [" + FLOW_FRAMEWORK_ENABLED.getKey() + "] to true.",
+                    RestStatus.FORBIDDEN
+                );
+            }
+
+            // Validate content
+            if (request.hasContent()) {
+                throw new FlowFrameworkException("No request body present", RestStatus.BAD_REQUEST);
+            }
+            // Validate params
+            String workflowId = request.param(WORKFLOW_ID);
+            if (workflowId == null) {
+                throw new FlowFrameworkException("workflow_id cannot be null", RestStatus.BAD_REQUEST);
+            }
+
+            boolean all = request.paramAsBoolean("all", false);
+            GetWorkflowRequest getWorkflowRequest = new GetWorkflowRequest(workflowId, all);
+            return channel -> client.execute(GetWorkflowAction.INSTANCE, getWorkflowRequest, ActionListener.wrap(response -> {
+                XContentBuilder builder = response.toXContent(channel.newBuilder(), ToXContent.EMPTY_PARAMS);
+                channel.sendResponse(new BytesRestResponse(RestStatus.OK, builder));
+            }, exception -> {
+                try {
+                    FlowFrameworkException ex = new FlowFrameworkException(exception.getMessage(), ExceptionsHelper.status(exception));
+                    XContentBuilder exceptionBuilder = ex.toXContent(channel.newErrorBuilder(), ToXContent.EMPTY_PARAMS);
+                    channel.sendResponse(new BytesRestResponse(ex.getRestStatus(), exceptionBuilder));
+                } catch (IOException e) {
+                    logger.error("Failed to send back provision workflow exception", e);
```

Comment:
nit - failed to send back get workflow exception?

Reply:
added